src/ext/h2c.js

```diff
@@ -233,13 +232,16 @@ function hashAndInc(seed, curve, hash) {
         h.update(ctrBits);
 
         const digestBits = h.finalize();
+        const bytes = sjcl.codec.bytes.fromBits(digestBits);
 
-        let point = decompressPoint(digestBits, curve, 0x02);
+        // attempt to decompress a point along with valid tags
+        // curve choice is implicit based on active curve parameters
+        let point = decompressPoint([2].concat(bytes));
         if (point !== null) {
             return point;
         }
 
-        point = decompressPoint(digestBits, curve, 0x03);
+        point = decompressPoint([3].concat(bytes));
```

Comment:
#### I noticed a logical error on this method.

Observe that trying to decode a point using `P = decode( 0x02 || bytes) ` is essentially the same as decoding `P = decode( 0x03 || bytes) `. Both points have the same `x` coordinate but different `y` coordinate. 
Then, if the first call succeeds, then the point in the second call is just the negative point obtained in the first call.
On the other hand, if the first call fails on decoding a point, the second call will fail as well. Hence it spent twice the time on every failure.